model/oauth/client.go

```diff
@@ -357,6 +368,15 @@ func (c *Client) Create(i *instance.Instance) *ClientRegistrationError {
 	c.GrantTypes = []string{"authorization_code", "refresh_token"}
 	c.ResponseTypes = []string{"code"}
 
+	// Adding Metadata
+	md := metadata.New()
+	if strings.HasPrefix(c.SoftwareID, "registry://") {
+		md.CreatedByApp = strings.TrimPrefix(c.SoftwareID, "registry://")
+	}
+	md.CreatedByAppVersion = c.SoftwareVersion
```

Comment:
We should move this line inside the if block: the `CreatedByAppVersion` is relevant only if `CreatedByApp` is set.